plugins/linux-sleep/fu-plugin-linux-sleep.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-hash.h"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+}
+
+void
+fu_plugin_add_security_attrs (FuPlugin *plugin, FuSecurityAttrs *attrs)
+{
+	gsize bufsz = 0;
+	g_autofree gchar *buf = NULL;
+	g_autoptr(FwupdSecurityAttr) attr = NULL;
+	g_autoptr(GError) error_local = NULL;
+	g_autoptr(GFile) file = g_file_new_for_path ("/sys/power/mem_sleep");
+
+	/* create attr */
+	attr = fwupd_security_attr_new ("org.kernel.CheckS3Sleep");
+	fwupd_security_attr_set_level (attr, FWUPD_SECURITY_ATTR_LEVEL_THEORETICAL);
+	fwupd_security_attr_set_name (attr, "Linux Kernel S3 Sleep");
+	fu_security_attrs_append (attrs, attr);
+
+	/* load file */
+	if (!g_file_load_contents (file, NULL, &buf, &bufsz, NULL, &error_local)) {
+		g_autofree gchar *fn = g_file_get_path (file);
+		g_warning ("could not open %s: %s", fn, error_local->message);
+		fwupd_security_attr_set_result (attr, "Deep sleep status unavailable");
+		return;
+	}
+	if (g_strstr_len (buf, bufsz, "deep") != NULL) {
```

Comment:
Didn't we say that we would only check what it was set to at daemon startup, not actually whether it was offered?

Reply:
I wasn't sure. The daemon doesn't have to be started at early-boot, it might be started by gnome-software well after the admin has done his "fixup". Maybe in this case the admin would just blacklist that one plugin?

Reply:
But so irrespective of when the daemon has actually been started this should be looking at "[deep]" instead.  And flagging that as "Suspend is set to S3"

Reply:
I just pushed what I'm thinking to the branch, see what you think.

Reply:
Works for me, thanks.